contracts/matic/SubscriptionManager.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin-upgradeable/contracts/access/AccessControlUpgradeable.sol";
+import "@openzeppelin-upgradeable/contracts/proxy/utils/Initializable.sol";
+
+contract SubscriptionManager is Initializable, AccessControlUpgradeable {
+
+    bytes32 public constant SET_RATE_ROLE = 
+        keccak256("Power to set the fee rate");
+    bytes32 public constant WITHDRAW_ROLE = 
+        keccak256("Power to withdraw funds from SubscriptionManager");
+
+    // The layout of policy struct is optimized, so sponsor, timestamps and size
+    // fit in a single 256-word.
+    struct Policy {
+        address payable sponsor;
+        uint32 startTimestamp;
+        uint32 endTimestamp;
+        uint16 size; // also known as `N`
+        // There's still 2 bytes available here
+        address owner;
+    }
+
+    event PolicyCreated(
```

Comment:
If we envision the feeRate possibly being changed, then should we include its value in the event?

Reply:
That's a good point. Maybe rather than the value of the feeRate, we should add here the total cost for the Policy. What do you think?

Reply:
Total cost of the policy sounds fine to me. 

@vzotova may have some thoughts here - the old PolicyCreated event had `feeRate` in it.

Reply:
In that case it had more sense since there was a potential variability depending on the minimum fee rate set by the nodes involved in the policy, while here we're changing the model so individual nodes/stakers can't unilaterally change the fee rate so `feeRate` will have probably not many changes.

Reply:
Right! 👍 . Total it is!

Reply:
I'm fine with any

Reply:
✅